app/src/main/java/com/matxowy/flashcardsapp/data/services/notifications/FirebaseNotificationService.kt

```diff
@@ -0,0 +1,68 @@
+package com.matxowy.flashcardsapp.data.services.notifications
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.NotificationManager.IMPORTANCE_HIGH
+import android.app.PendingIntent
+import android.app.PendingIntent.FLAG_IMMUTABLE
+import android.app.PendingIntent.FLAG_ONE_SHOT
+import android.content.Context
+import android.content.Intent
+import android.graphics.Color
+import androidx.core.app.NotificationCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.matxowy.flashcardsapp.R
+import com.matxowy.flashcardsapp.app.MainActivity
+import com.matxowy.flashcardsapp.data.localpreferences.LocalPreferencesRepository
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import kotlin.random.Random
+
+@AndroidEntryPoint
+class FirebaseNotificationService : FirebaseMessagingService() {
+    @Inject
+    lateinit var localPreferencesRepository: LocalPreferencesRepository
+
+    override fun onMessageReceived(message: RemoteMessage) {
+        super.onMessageReceived(message)
+
+        val intent = Intent(this, MainActivity::class.java)
+        val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val notificationId = Random.nextInt()
+
+        createNotificationChannel(notificationManager)
+
+        val pendingIntent = PendingIntent.getActivity(this, REQUEST_CODE, intent, FLAG_IMMUTABLE or FLAG_ONE_SHOT)
+        val notification = NotificationCompat.Builder(this, CHANNEL_ID)
+            .setContentTitle(message.notification?.title)
+            .setContentText(message.notification?.body)
+            .setSmallIcon(R.drawable.ic_launcher_foreground)
+            .setAutoCancel(true)
+            .setContentIntent(pendingIntent)
+            .build()
+
+        notificationManager.notify(notificationId, notification)
+    }
+
+    override fun onNewToken(token: String) {
+        super.onNewToken(token)
+        localPreferencesRepository.saveUserNotificationToken(token)
```

Comment:
W jakim celu jest zapisywany ten token do shared preferences? Nie widzę, żeby był gdzieś później odczytywany.

Reply:
To jest tylko przygotowanie pod ewentualną dalszą rozbudowę tej funkcjonalności. Na ten moment nigdzie go nie odczytujemy, ale zauważyłem warning, że dobrze by było dodać tą metodę onNewToken to dodałem i od razu uznałem, że na przyszłość zapiszę ten token ☺️